src/OpenTelemetry/Trace/Configuration/OpenTelemetrySdk.cs

```diff
@@ -92,12 +94,22 @@ public static OpenTelemetrySdk EnableOpenTelemetry(Action<OpenTelemetryBuilder>
                 }
             }
 
+            openTelemetrySDK.resource = openTelemetryBuilder.Resource;
+
             // This is what subscribes to Activities.
             // Think of this as the replacement for DiagnosticListener.AllListeners.Subscribe(onNext => diagnosticListener.Subscribe(..));
             openTelemetrySDK.listener = new ActivityListener
             {
                 // Callback when Activity is started.
-                ActivityStarted = activityProcessor.OnStart,
+                ActivityStarted = (activity) =>
+                {
+                    if (activity.IsAllDataRequested)
+                    {
+                        activity.SetCustomProperty("Resource", openTelemetrySDK.resource);
```

Comment:
This seems to be a considerable overhead.

The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/27e5c89075d3eeb5981e79bf8ab94d78aea5481d/specification/resource/sdk.md#resource-operations) doesn't require resource to be attached to span, and I am guessing resource is like global to the SDK?

Reply:
Yea my implementation above is bad.
Resource is not global - its associated with TracerProvider and all Tracer (and Spans from it) from this provider will be associated with that resource.
Will check whats the best way to achieve this.

Reply:
Checking the perf impact of storing Resource inside every Activity. It could be same as adding just another Tag for sampled-in Activity.
If there are no perf concerns, will proceed with this approach. Else need to find alternates. 
(If anyone has better ideas, please do share)

Reply:
--without resource populating inside Activity
```
|                                                   Method |     Mean |     Error |    StdDev |   Median |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------------------------------------------- |---------:|----------:|----------:|---------:|-------:|------:|------:|----------:|
|              CreateActivity_WithAttributes_NoOpProcessor | 1.184 us | 0.0266 us | 0.0567 us | 1.163 us | 0.2899 |     - |     - |   1.19 KB |
| CreateActivity_WithAttributesAndCustomProp_NoOpProcessor | 1.443 us | 0.0443 us | 0.1248 us | 1.392 us | 0.4559 |     - |     - |   1.87 KB |
```

--with resource populated inside CustomProperty of Activity

```
|                                                   Method |     Mean |     Error |    StdDev |   Median |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------------------------------------------- |---------:|----------:|----------:|---------:|-------:|------:|------:|----------:|
|              CreateActivity_WithAttributes_NoOpProcessor | 1.390 us | 0.0284 us | 0.0533 us | 1.366 us | 0.4654 |     - |     - |   1.91 KB |
| CreateActivity_WithAttributesAndCustomProp_NoOpProcessor | 1.421 us | 0.0282 us | 0.0356 us | 1.407 us | 0.4673 |     - |     - |   1.91 KB |
```

Reply:
As shown from benchmark result above, putting Resource into customproperty has only minor/negligible impact.
(There is visible impact of using Activity.Tags only vs using Tags and CustomProperty - this is a separate issue. Will check with .NET team if this can be improved)

Reply:
@cijothomas If someone had multiple pipelines, with more than one ActivityListener, potentially with different resources, would they step on each other?

Reply:
And shorter name might make a small perf improvement due to the reduced hashing cost :)

BTW, we probably need to reserve a namespace for custom properties and have some naming conventions.

Reply:
@CodeBlanch yes its possible to have multiple `OpenTelemetrySdk` each listening to the same `ActivitySource` - the activity's Resource will be overwritten by the last one.

In Span based API, it was not possible to have this situation - If one creates multiple `TracerProvider`, the `Tracer` from them will be independent and their `Span`s will be independent as well. Need to think more on this and propose a solution.

Reply:
For users who want to have multiple pipeline for same ActivitySource, they can do so without instantiating new `OpenTelemetrySdk` instance. Still wondering whats the use case for multiple instances of this, unless each one listens to distinct ActivitySources.

Reply:
Not sure what the use case would be. In my real-world usage of the library, I haven't run into a situation where I would need multiple pipelines yet. I have at different times had one pipeline exporting to multiple things (Zipkin, Jaeger, Stackdriver). Any way I only brought it up because right now the code allows you to do multiple so IMO we should support it or make it not allowed 🙅

Reply:
I can fully understand the need of multiple pipeline - one with AlwaysOnSampler and NonExpensiveExporter, and another with ProbabilitySampler(5%) and ExpensiveExporter. (Inside Microsoft a lot of teams does this already.)

What I couldn't yet figure out is : Why would one want to have multiple OpenTelemetrySdk instance, for the same ActivitySources.

I will open an issue to track this. Agree we need a way to either allow it and explain the use case, or stop allowing it.